java/vector/src/main/java/org/apache/arrow/vector/complex/ListViewVector.java

```diff
@@ -0,0 +1,913 @@
+   *
+   * @return the value capacity
+   */
+  @Override
+  public int getValueCapacity() {
+    return getValidityAndOffsetValueCapacity();
+  }
+
+  private int getValidityAndSizeValueCapacity() {
+    final int offsetValueCapacity = Math.max(getOffsetBufferValueCapacity(), 0);
+    final int sizeValueCapacity = Math.max(getSizeBufferValueCapacity(), 0);
+    return Math.min(offsetValueCapacity, sizeValueCapacity);
+  }
+
+  private int getValidityAndOffsetValueCapacity() {
+    final int offsetValueCapacity = Math.max(getOffsetBufferValueCapacity(), 0);
+    return Math.min(offsetValueCapacity, getValidityBufferValueCapacity());
+  }
+
+  private int getValidityBufferValueCapacity() {
+    return capAtMaxInt(validityBuffer.capacity() * 8);
+  }
+
+  /**
+   * Set the element at the given index to null.
+   * @param index the value to change
+   */
+  @Override
+  public void setNull(int index) {
+    while (index >= getValidityAndSizeValueCapacity()) {
+      reallocValidityAndSizeAndOffsetBuffers();
+    }
+    if (lastSet >= index) {
+      lastSet = index - 1;
+    }
+
+    if (index == 0) {
+      offsetBuffer.setInt(0, 0);
+      sizeBuffer.setInt(0, 0);
+    } else {
+      final int prevOffset = offsetBuffer.getInt((index - 1) * OFFSET_WIDTH);
+      final int prevSize = sizeBuffer.getInt((index - 1) * SIZE_WIDTH);
+      final int currOffSet = prevOffset + prevSize;
+      offsetBuffer.setInt(index * OFFSET_WIDTH, currOffSet);
+      sizeBuffer.setInt(index * SIZE_WIDTH, 0);
+    }
+
+    BitVectorHelper.unsetBit(validityBuffer, index);
+    lastSet = index;
+  }
+
+  /**
+   * Start new value in the ListView vector.
+   * There are a few cases that are handled in this function.
+   * There are two main scenarios that need to be considered.
+   * The first scenario is simple insertion where indices are continuously updated.
+   * The other scenario is the event of non-continuous writing,
+   * the offset buffer needs to be updated.
+   *
+   * @param index index of the value to start
+   * @return offset of the new value
+   */
+  @Override
+  public int startNewValue(int index) {
+    while (index >= getValidityAndSizeValueCapacity()) {
+      reallocValidityAndSizeAndOffsetBuffers();
+    }
+
+    if (lastSet >= index) {
+      lastSet = index - 1;
+    }
+
+    if (index == 0) {
+      offsetBuffer.setInt(0, 0);
+    } else if (index > lastSet) {
+      /* when skipping indices, we need to update the offset buffer */
+      /* setting offset from lastSet + 1 to index (included) */
+      for (int i = lastSet + 1; i <= index; i++) {
+        if (i == 0) {
+          offsetBuffer.setInt(0, 0);
+          continue;
+        }
+        final int prevOffSet = offsetBuffer.getInt((i - 1L) * OFFSET_WIDTH);
+        final int prevSize = sizeBuffer.getInt((i - 1L) * SIZE_WIDTH);
+        final int currOffSet = prevOffSet + prevSize;
+        offsetBuffer.setInt(i * OFFSET_WIDTH, currOffSet);
+      }
```

Comment:
shouldn't we be setting the size here...?

Reply:
@lidavidm I updated the logic. Now there is no `lastSet` as you initially thought. Could you please check the logic. Also I think we should only set the size buffer once we know the length of the data appended to a particular list at a given index. That can be only known when we call the `endList()` function. Please correct me if I am wrong or misunderstood your question.